### ISSUE TITLE
STP /STEP Inport error

### ISSUE THREAD

**Issue by Dade1984** [bambulab/BambuStudio#5239]
### Bambu Studio Version

10.10.0.89

### Where is the application from?

Bambu Lab Official website

### OS version

Windows 11

### Printer

Bambu x1c

### How to reproduce

STP /STEP Import error
software close

### Actual results

. STP /STEP Import error
software close

### Expected results

. STP /STEP Import error
software close

### Project file & Debug log uploads

STP /STEP Import error
software close

### Checklist of files to include

- [X] Log file
- [X] Project file

**Comment by Stevelsch** [comment]
Related to https://github.com/bambulab/BambuStudio/issues/5261

**Comment by DoTsTeR77** [comment]
yes, major issues with me importing step files that previously imported fine.. the beta had the same problem.. I had to revert back to previous version..

**Comment by MartinChan3** [comment]
It seems that Studio has released a temporary package to provide an interim solution for those experiencing issues on Mac (Intel chip), STEP import crashes, and missing temperature displays. Hopefully, they will release the actual hotfix soon!
Windows：https://github.com/bambulab/BambuStudio/releases/download/v01.10.00.89/Bambu_Studio_win_public-v01.10.00.94-20241114223322.exe
Mac：https://github.com/bambulab/BambuStudio/releases/download/v01.10.00.89/Bambu_Studio_mac-v01.10.00.94-20241114223322.dmg

**Comment by DoTsTeR77** [comment]
yup, just tried the new  package.. I still cant import some step files no matter how much I change those sliders.. I wish they would just put it back the way it was.. or put an option to import them the way the previous version did which gave me no issues..

**Comment by DoTsTeR77** [comment]
> It seems that Studio has released a temporary package to provide an interim solution for those experiencing issues on Mac (Intel chip), STEP import crashes, and missing temperature displays. Hopefully, they will release the actual hotfix soon! Windows：https://github.com/bambulab/BambuStudio/releases/download/v01.10.00.89/Bambu_Studio_win_public-v01.10.00.94-20241114223322.exe Mac：https://github.com/bambulab/BambuStudio/releases/download/v01.10.00.89/Bambu_Studio_mac-v01.10.00.94-20241114223322.dmg

yup, just tried the new package.. I still cant import some step files no matter how much I change those sliders.. I'd love to see it back the way it was.. or put an option to import them the way the previous version did which gave me no issues..

Unfortunately I cant provide the step files as it work under NDA..

ohh.. Could I ask what those sliders values would have been by default if the previous version used them?

Also.. all work on this is appreciated.. I dont want to sound like it isnt..

**Comment by MackBambu** [comment]
> ohh.. Could I ask what those sliders values would have been by default if the previous version used them?

liner deflection is 0.03, angle deflection is 0.5.
All algorithms remain unchanged; in fact, only the parameter settings have been made configurable.

**Comment by Hardrada** [comment]
This issue is making the app completely and utterly useless and it should be prioritized as an immediate fix. The software literally depends on step file imports and thus this should be the number one issue fixed. I've had to use orca slicer to get my prints done and then I can't import them back into bambu studio because of a file version mis-match.. This is killing me.

**Comment by DoTsTeR77** [comment]
I second that issue in not ok and needs to be taken care of..   the step file import process needs to revert back to version 1.97.52 behavior.. it worked fine under that version..

**Comment by DoTsTeR77** [comment]
> @Hardrada @DoTsTeR77 Deleting the configuration file will reset the parameters, but the logic for using the parameters remains unchanged from the previous version. I will wait for the feedback because I still don't know what happened.
> 
> > I second that issue in not ok and needs to be taken care of.. the step file import process needs to revert back to version 1.9,7.52 behavior.. it worked fine under that version..
> 
> I suspect you may have set an incorrect parameter that prevents successful meshing and enabled "Do not show again."

you suspect wrong.

**Comment by MackBambu** [comment]
> not the case
@DoTsTeR77 
Would it be possible for you to record an operation video? This would help me reproduce the issue. Thank you.

**Comment by nitrousdave** [comment]
I too have the same problem Hardrada does when trying to use a step file as a modifier. This was only after the file import dialog was added. Windows 10 and 11 computers both have the problem. Seems like the step file import dialog should pop up when using a step as a modifier.

**Comment by Stevelsch** [comment]
Yes. The step import is ruined by this update. Still does not work. - and why do this change at all?Sent from my iPhoneOn 3 Dec 2024, at 20.25, nitrousdave ***@***.***> wrote:﻿
I too have the same problem Hardrada does when trying to use a step file as a modifier. This was only after the file import dialog was added. Windows 10 and 11 computers both have the problem. Seems like the step file import dialog should pop up when using a step as a modifier.

**Comment by MackBambu** [comment]
The new version has already switched to a different threading model. Does the issue still persist?

**Comment by stvelsc** [comment]
It is supposed to look like this:

<img width="858" alt="Image" src="https://github.com/user-attachments/assets/cf5dc823-93f8-4373-9871-ed2bb5a368f4" />

**Comment by MackBambu** [comment]
> Just tested with the same part:
> [Shell_Part.txt](https://github.com/user-attachments/files/19952374/Shell_Part.txt)

Try reducing the linear deflection and angle deflection to achieve more subdivisions. It seems that 0.002 and 0.05 produce better results.
![Image](https://github.com/user-attachments/assets/87b565a5-786e-443d-ba76-8a4dd5d6bc6c)

![Image](https://github.com/user-attachments/assets/84b746ee-4bac-43df-8de9-a12244346df0)

**Comment by stvelsc** [comment]
Wow. Great thank you very much! I am back to using STEP then!
